pkg/corerp/api/v20231001preview/environment_conversion_test.go

```diff
@@ -545,3 +572,315 @@ func getTestKubernetesEmptyMetadataExtensions(t *testing.T) []datamodel.Extensio
 
 	return extensions
 }
+
+func Test_toRecipeConfigTerraformProvidersDatamodel(t *testing.T) {
+	tests := []struct {
+		name   string
+		config *RecipeConfigProperties
+		want   map[string][]datamodel.ProviderConfigProperties
+	}{
+		{
+			name:   "Empty Recipe Configuration",
+			config: &RecipeConfigProperties{},
+			want:   nil,
+		},
+		{
+			name: "Single Provider Configuration",
+			config: &RecipeConfigProperties{
+				Terraform: &TerraformConfigProperties{
+					Providers: map[string][]map[string]any{
+						"azurerm": {
+							{
+								"subscription_id": "00000000-0000-0000-0000-000000000000",
+							},
+						},
+					},
+				},
+			},
+			want: map[string][]datamodel.ProviderConfigProperties{
+				"azurerm": {
+					{
+						AdditionalProperties: map[string]any{
+							"subscription_id": "00000000-0000-0000-0000-000000000000",
+						},
+					},
+				},
+			},
+		},
+		{
+			name: "Single Provider With Multiple Configuration",
+			config: &RecipeConfigProperties{
+				Terraform: &TerraformConfigProperties{
+					Providers: map[string][]map[string]any{
+						"azurerm": {
+							{
+								"subscription_id": "00000000-0000-0000-0000-000000000000",
+							},
+							{
+								"tenant_id": "00000000-0000-0000-0000-000000000000",
```

Comment:
Let's use a more realistic scenario where an alias is added. Multiple configurations for same provider without an alias isn't something we expect users to be setting or encourage to set since that will just result into last config overriding the rest.

Reply:
I updated this one as discussed offline.